docs/DeveloperGuide.md

```diff
@@ -1,29 +1,156 @@
 # Developer Guide
+The aim of this guide is to help readers understand how the system and components of RecipEditor 
+is designed, implemented and tested. This developer guide also serves to help 
+developers to understand the architecture of RecipEditor and some design considerations. 
+Click to view the latest release of [RecipEditor]((https://github.com/AY2223S1-CS2113-T18-2/tp/releases)).
+
+## Content page
+[Acknowledgements](#acknowledgements)
+
+[Design](#design)
+- [Architecture](#architecture)
+- [Logic Component](#logic-component)
+- [Ui Component](#ui-component)
+- [Storage Component](#storage-component)
+- [Implementation](#implementation)
```

Comment:
Should implementation be here? Or as a sibling or the Design section?
Ask these questions for your other sections as well(product scope/user-stories)